utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -386,10 +393,15 @@ impl<'trie, T: TrieValue> CodePointTrie<'trie, T> {
         P: TrieValue,
     {
         let converted_data = self.data.try_into_converted()?;
+        let error_ule = self.error_value.to_unaligned();
+        let slice = &[error_ule];
+        let error_vec = ZeroVec::<T>::Borrowed(slice);
+        let error_converted = error_vec.try_into_converted::<P>()?;
         Ok(CodePointTrie {
             header: self.header,
             index: self.index,
             data: converted_data,
+            error_value: error_converted.get(0).unwrap_or(P::DEFAULT_ERROR_VALUE),
```

Comment:
Nit: We really would like to get rid of `DEFAULT_ERROR_VALUE`; that is the main purpose of the ticket. This PR moves around validation logic but doesn't get rid of `DEFAULT_ERROR_VALUE`, and instead uses it in _even more esoteric_ places. I think we should

1. Add `#[allow(unwrap_used)]` with a good comment explaining why this case is impossible
2. Open an issue to explore single-value validation in ZeroVec, which is something I've kind-of wanted to explore for a while anyway